internal/registry/types.bucket.go

```diff
@@ -362,28 +346,29 @@ func (b *Bucket) PopulateIteration(ctx context.Context) error {
 		}
 	}
 
+	if len(toCreate) == 0 {
+		return nil
+	}
+
 	var errs *multierror.Error
 	var wg sync.WaitGroup
-
 	for _, buildName := range toCreate {
 		wg.Add(1)
 		go func(name string) {
 			defer wg.Done()
+
 			log.Printf("[TRACE] registering build with iteration for %q.", name)
-			// Need a way to handle skipping builds that were already created.
-			// TODO when we load an existing iteration we will probably have a build Id so we should skip.
-			// we also need to bubble up the errors here.
 			err := b.CreateInitialBuildForIteration(ctx, name)
+
 			if checkErrorCode(err, codes.AlreadyExists) {
-				// Check whether build is complete, and if so, skip it.
 				log.Printf("[TRACE] build %s already exists in Packer registry, continuing...", name)
 				return
 			}
 
 			errs = multierror.Append(errs, err)
 		}(buildName)
+		wg.Wait()
```

Comment:
Why did this one move inside the for?

Reply:
This was moved to fix a race condition that is introduced by calling CreateBuild on the MockService. By spawning off multiple routines for the CreateBuild the Mock Data is overwritten before it can be read resulting in a data race. This change makes it a bit more sequential. Maybe the correct change should be to leave it as it as and add a mutex to the Mock Client to prevent overwrites.

Reply:
So I updated the Mock service a little to be a bit more concurrent. It is a hack TBH but it works for the testing purposes. In doing so I found a race condition with the appending of errors so that it good.